KoderLabs/ngx-device-detector
# chore: update to Angular 15

Hello,

This PR does:

- Closes https://github.com/KoderLabs/ngx-device-detector/issues/257
- Update to latest Angular 14 packages
- Update all packages to latest version except:
  - `typescript`
  - `@types/node`
  - `jest`
    - `jest-preset-angular` is not yet compatible with `jest 29`
- Update the lib to only support Angular `^15`
- Update the app to Angular 15
  - `browserlistrc` was removed
  - `environment` files have been deleted
    - `angular.json` was update to reflect that
- I removed `ngx-device-detector` from the installed packages, it now use the one built locally
  - It prevented from updating to Angular 15 (I don't like `--force`)
  - The lib should be built before launching the demo app
  - Use the `paths` entry in the `tsconfig.json` file
- I updated your prettier scripts
  - Now named `format:check` and `format:all`
  - It now prettier all projects and `.ts,.js,.json,.html,.scss}` files
- All files were formatted
- Only update patch package versions

Feel free to also remove other PR/branche related to packages update for cleaning purpose.

If everything is ok, I can make a new commit to bump the version.
Could you release a new version following this PR? (This would be much appreciated heh)

Have a good day.

cc @AhsanAyaz
